src/librustdoc/clean/mod.rs

```diff
@@ -448,10 +415,60 @@ impl Clean<Item> for doctree::Module {
     }
 }
 
+pub trait Attributes {
+    fn has_word(&self, &str) -> bool;
+    fn value<'a>(&'a self, &str) -> Option<&'a str>;
+    fn list<'a>(&'a self, &str) -> Option<&'a [Attribute]>;
+    fn list_def<'a>(&'a self, &str) -> &'a [Attribute];
```

Comment:
Maybe this is the only needed method for rustdoc? It looks like `list` may not be used all that often